src/components/DesignSystemButton/DesignSystemButton.scss

```diff
@@ -0,0 +1,120 @@
+@use '../../colours.scss' as *;
+
+:root * {
+  --rpf-button-primary-background-color: #{$rpf-teal-primary-default};
+  --rpf-button-primary-background-color-hover: #{$rpf-teal-primary-hover};
+  --rpf-button-primary-background-color-active: #{$rpf-teal-primary-active};
+  --rpf-button-primary-text-color: #{$rpf-text-primary};
+
+  .--dark * {
+    --rpf-button-primary-background-color: #{$rpf-teal-primary-default-darkmode};
+    --rpf-button-primary-background-color-hover: #{$rpf-teal-primary-hover-darkmode};
+    --rpf-button-primary-background-color-active: #{$rpf-teal-primary-active-darkmode};
+  }
+}
+
+.rpf-button--primary * {
+  svg {
+    fill: $rpf-text-primary;
+  }
+}
+
+.rpf-button--secondary {
+  justify-content: center;
+  align-items: center;
+  flex-direction: row-reverse;
+}
+
+.rpf-button--secondary * {
+  svg {
+    fill: $rpf-text-secondary;
+  }
+}
+
+.--dark * {
+  .rpf-button--secondary {
+    border-color: $rpf-teal-400;
+
+    span {
+      color: $rpf-white;
+    }
+
+    svg {
+      fill: $rpf-white;
+    }
+
+    &::before {
+      background-color: rgba($rpf-grey-700, 1);
+    }
+
+    &:hover {
+      &::before {
+        background-color: rgba($rpf-white, 0.1);
+      }
+
+      border-color: $rpf-teal-400;
+    }
+
+    &:active {
+      &::before {
+        background-color: rgba($rpf-white, 0.15);
+      }
+    }
+
+    &:disabled {
+      &::before {
+        background-color: inherit;
+      }
+
+      span {
+        color: $rpf-grey-150;
+      }
+
+      border-color: $rpf-grey-150;
+      background-color: inherit;
+    }
+  }
+}
+
+.--light * {
+  .rpf-button--secondary {
+    span {
+      color: $rpf-text-primary;
+    }
+  }
+
+  .rpf-button--secondary::before {
+    background-color: inherit;
+  }
+
+  &::before {
+    background-color: rgba($rpf-white, 1);
+  }
+
+  &:hover {
+    &::before {
+      background-color: rgba(231, 248, 247, 1);
+    }
+
+    border-color: $rpf-teal-400;
+  }
+
+  &:active {
+    &::before {
+      background-color: rgba(220, 245, 243, 1);
+    }
+  }
+
+  &:disabled {
+    &::before {
+      background-color: inherit;
+    }
+
+    span {
+      color: rgba(213, 215, 220, 1);
+    }
+
+    border-color: rgba(213, 215, 220, 1);
+    background-color: inherit;
+  }
+}
```

Comment:
☝️ I REALLY like this approach. It was quick and dirty to put them into the `App.scss` but this is truly expandable and gives us a bunch more control and options 💯